### PR TITLE
Fixes #237 - Upgrade to elasticsearch 7.3.1

### LINKED ISSUE #237: Elasticsearch 7.3
Since ES 7.3 as been released, it would be nice if the plugin supported it.

### PR DESCRIPTION
**Pull request by iagodvsantos**
- Rebuild gradle wrapper
- Upgrade ES to 7.3.1

Note: Since there is an error on `build-tools` es plugin in the version 7.3.1 as stated here -> https://github.com/elastic/elasticsearch/issues/45891#issuecomment-525399411 a workaround was necessary 

Closes:. https://github.com/o19s/elasticsearch-learning-to-rank/issues/237

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -51,10 +51,10 @@ dependencies {
   compile 'org.ow2.asm:asm:5.0.4'
   compile 'org.ow2.asm:asm-commons:5.0.4'
   compile 'org.ow2.asm:asm-tree:5.0.4'
-  compile 'org.elasticsearch:elasticsearch:7.2.0'
+  compile 'org.elasticsearch:elasticsearch:7.3.0'
```
Comment (nomoa) [inline build.gradle]: we need to bump org.apache.lucene:lucene-expressions:8.0.0 to 8.1.0 (c.f. https://github.com/elastic/elasticsearch/blob/v7.3.0/buildSrc/version.properties#L2)

**Comment by jillesvangurp** [comment]
7.3.1 is out by now.

File: build.gradle (hunk 1 above)
Comment (iagodvsantos) [reply build.gradle]: Done!

**Comment by jillesvangurp** [comment]
It would help us if we could upgrade this to 7.3.1. I think this should not introduce additional api changes.

**Comment by softwaredoug** [comment]
Thanks @iagodvsantos @nomoa @jillesvangurp 

All looks good, but when rebuilding locally I got the errors:

```
> Task :test
Exception in thread "Thread-3" java.lang.ClassCastException: class org.apache.logging.log4j.simple.SimpleLoggerContext cannot be cast to class org.apache.logging.log4j.core.LoggerContext (org.apache.logging.log4j.simple.SimpleLoggerContext and org.apache.logging.log4j.core.LoggerContext are in unnamed module of loader 'app')
        at org.elasticsearch.test.ESTestCase.lambda$static$0(ESTestCase.java:236)
        at java.base/java.lang.Thread.run(Thread.java:835)
        Suppressed: java.lang.IllegalStateException: No context information for thread: Thread[id=26, name=Thread-3, state=RUNNABLE, group=TGRP-ValidatingLtrQueryBuilderTests]. Is this thread running under a class com.carrotsearch.randomizedtesting.RandomizedRunner runner context? Add @RunWith(class com.carrotsearch.randomizedtesting.RandomizedRunner.class) to your test class. Make sure your code accesses random contexts within @BeforeClass and @AfterClass boundary (for example, static test class initializers are not permitted to access random contexts).
                at com.carrotsearch.randomizedtesting.RandomizedContext.context(RandomizedContext.java:249)
                at com.carrotsearch.randomizedtesting.RandomizedContext.current(RandomizedContext.java:134)
                at com.carrotsearch.randomizedtesting.RandomizedRunner.augmentStackTrace(RandomizedRunner.java:1885)
                at com.carrotsearch.randomizedtesting.RunnerThreadGroup.uncaughtException(RunnerThreadGroup.java:20)
                at java.base/java.lang.Thread.dispatchUncaughtException(Thread.java:1997)

BUILD SUCCESSFUL in 1m 23s
```

**Comment by jillesvangurp** [comment]
I've been running with the 7.3.1 build today and did not notice this issue. But there may be some issue with the test classpath dependencies for log4j. It may be pulling in different versions for log4j-core and log4j-api transitively. I've seen similar issues on other plugin projects. I think the api dependency might be coming along with elasticsearch.

In any case, the solution is adding matching versions of both as test dependencies.

**Comment by jillesvangurp** [comment]
I was not able to reproduce the log related exception. It produces no output during gradle test. But if this comes back; try adding these dependencies.

```
testCompile 'org.apache.logging.log4j:log4j-core:2.12.1'
testCompile 'org.apache.logging.log4j:log4j-api:2.12.1'
```

**Pull request merged**